tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -852,7 +870,44 @@ const reducer = createReducer(
         }
       }
 
-      const nextState = {
+      if (response.runToSeries) {
+        for (let runId in response.runToSeries) {
+          const cardMetadata: CardMetadata = buildCardMetata({
+            plugin,
+            tag,
+            runId,
+          });
+          if (sample) {
+            cardMetadata.sample = sample;
+          }
+
+          const cardId = getCardId(cardMetadata);
+          const cardTimeSeriesLoadable = getTimeSeriesLoadable(
+            nextTimeSeriesData,
+            cardMetadata.plugin,
+            cardMetadata.tag,
+            cardMetadata.sample
+          );
+          if (cardTimeSeriesLoadable) {
+            const nextMinMax = generateCardMinMaxStep(
+              cardTimeSeriesLoadable.runToSeries
```

Comment:
I still prefer to do it upfront. While the complexity is adding up it will at least all be upfront and in one place.

You are right, this did not correctly calculate the min/max for histograms and images cards, I didn't realize there each only contained a single run. 
I've updated this to only read the data from a single run for each of those card types.

Image cards are more complicated. I believe this is actually right for them despite the fact that the number of image cards is Tags * Runs * Samples the data only seems to be Tags * Runs.